pyvista/plotting/axes_actor.py

```diff
@@ -1,176 +1,416 @@
-        self.z_axis_shaft_properties.opacity = pyvista.global_theme.axes.z_color.int_rgba[3]
-        self.z_axis_tip_properties.opacity = pyvista.global_theme.axes.z_color.int_rgba[3]
-        self.z_axis_shaft_properties.lighting = pyvista.global_theme.lighting
+        self.__enable_orientation_workaround = True
+
+        # Supported aliases (inherited from `create_axes_actor`)
+        x_label = kwargs.pop('xlabel', x_label)
+        y_label = kwargs.pop('ylabel', y_label)
+        z_label = kwargs.pop('zlabel', z_label)
+
+        # Deprecated on v0.43.0, completely removed in version 0.46.0
+        cone_radius = kwargs.pop('cone_radius', False)
+        if cone_radius:  # pragma: no cover
+            warnings.warn(
+                "Use of `cone_radius` is deprecated and will be removed in 0.46.0. "
+                "Use `tip_radius` instead.",
+                PyVistaDeprecationWarning,
+            )
+            tip_radius = cone_radius
+
+        # Deprecated on v0.43.0, completely removed in version 0.46.0
+        line_width = kwargs.pop('line_width', False)
+        if line_width:  # pragma: no cover
+            warnings.warn(
+                "Use of `line_width` is deprecated and will be removed in 0.46.0. "
+                "Use `shaft_width` instead.",
+                PyVistaDeprecationWarning,
+            )
+            shaft_width = line_width
+
+        if properties is None:
+            properties = dict()
+        if not isinstance(properties, dict):
+            raise TypeError("Properties must be a dictionary.")
+
+        # Deprecated on v0.43.0, completely removed in version 0.46.0
+        ambient = kwargs.pop('ambient', False)
+        if ambient:  # pragma: no cover
+            warnings.warn(
+                f"Use of `ambient` is deprecated and will be removed in 0.46.0. "
+                f"Use `properties={{'ambient':{ambient}}}` instead.",
+                PyVistaDeprecationWarning,
+            )
+            properties.setdefault('ambient', ambient)
+
+        # Set axis shaft and tip properties
+        properties.setdefault('lighting', pyvista.global_theme.lighting)
+        if len(properties) > 0:
+            x_shaft = self.x_axis_shaft_properties
+            y_shaft = self.y_axis_shaft_properties
+            z_shaft = self.z_axis_shaft_properties
+            x_tip = self.x_axis_tip_properties
+            y_tip = self.y_axis_tip_properties
+            z_tip = self.z_axis_tip_properties
+            for name, value in properties.items():
+                setattr(x_shaft, name, value)
+                setattr(y_shaft, name, value)
+                setattr(z_shaft, name, value)
+                setattr(x_tip, name, value)
+                setattr(y_tip, name, value)
+                setattr(z_tip, name, value)
+
+        if len(kwargs) > 0:
+            raise TypeError(
+                f"AxesActor() got an unexpected keyword argument '{list(kwargs.keys())[0]}'"
+            )
```

Comment:
Please use `assert_empty_kwargs` for this. 

https://github.com/pyvista/pyvista/blob/76aa761dac8dddf9995a49872399e4f9a406ecf8/pyvista/core/utilities/misc.py#L12